### ISSUE TITLE
Emails are incorrectly sanitised on signup

### ISSUE THREAD

**Issue by tblacker7** [buddyboss/buddyboss-platform#2270]
When using the BuddyBoss registration routine (/register), if a user has an apostrophe in their email address (allowed as per https://tools.ietf.org/html/rfc369), BuddyBoss rewrites this. So, for example tony.de'ath@domain.com becomes tony.de&ampamp#039ath@domain.com. This then means that the confirmation email is never sent, and the user can never interact with the site. We also then have emails going out to incorrect (non existent) email addresses.

**To Reproduce**
In admin > BuddyBoss > Settings, check "Allow non-members to register new accounts" to be active, set Registration Form to "BuddyBoss Registration". In an incognito window, create a new account with an email containing an apostrophe in the name before the @. After completing, in admin navigate to Users > Pending. You will see the email address has been modified.

**Expected behavior**
The email address should not be modified, simply checked for validity and an error thrown if invalid.

**Screenshots**
![Screenshot 2021-03-24 at 11 51 11](https://user-images.githubusercontent.com/44897304/112311580-c357d700-8c9d-11eb-92cc-6e222409d7c1.png)
The first account was registered via BuddyBoss registration page. You can see the malformed email address. The second account was registered using WordPress native registration, and you can see this email address is untouched.

**Cause**
The registration routine includes a function bp_members_signup_sanitization() in buddyboss-platform/bp-members/bp-members-filters.php lines 33 to 55 which takes all signup fields and runs them through the following sequence:
```
// Add the filters to each field.
	foreach ( $fields as $filter ) {
		add_filter( $filter, 'esc_html', 1 );
		add_filter( $filter, 'wp_filter_kses', 2 );
		add_filter( $filter, 'stripslashes', 3 );
	}
```
This is NOT needed for the email address, as it should already be sanitised using WP native logic. 
esc_html will convert the apostrophe to `&#039;` EVERY TIME.

This function should not include bp_get_signup_email_value, or it should be specifically excluded from the filters

**Comment by mahedihasannoman** [comment]
Hi @tblacker7 , apostrophe in email is not supported in BuddyBoss Platform.

Yes, you can indeed include an apostrophe or single quote in your e-mail address. You can even include other characters, like the % and even blank spaces. However, e-mail providers don’t always recognize nonalphanumeric characters.

Having too many nonstandard characters will confuse other people. For these reasons, it’s best to avoid apostrophes and other nonalphanumeric characters in your e-mail address unless you have a very good reason to include them.

That's why we do not allow apostrophe in email.

**Comment by tblacker7** [comment]
Not overly impressed with this being closed to be honest. Whilst I agree apostrophes and other alpha numeric characters are possibly not the best idea, some email providers - and in particular work place email addresses - DO allow apostrophes. In our case, this is not uncommon at all, and is a barrier to us actually using this tool as it could be viewed as discriminatory against certain races and ethnicities, where apostrophes are more common in peoples names.

In summary, we are NOT the controllers of the email addresses people already have, and therefore this seems like an arbitrary barrier (which is relatively easily solved in code) which is completely unnecessary.

**Comment by mahedihasannoman** [comment]
Hi @tblacker7 , I am reopening this issue. We will fix it soon.

**Comment by chetansatasiya** [comment]
@mahedihasannoman We can't make this change at a product level and it's a best PHP practice to sanitize and escape the email address. You can provide the snippet to customers to remove this filter on their end in their child theme functions file.

As I am seeing Gmail, Yahoo, and also outlook will not give you permission to create those kinds of email addresses.

I am closing this issue and feel free to comment here if you have any questions on this.

CC @dipskakadiya @mikejyoung @sourabhmatolia @KartikSuthar

**Comment by singhleo** [comment]
Note - 
tony.de'ath@domain.com is allowed by is_email function. 

# Solutions I propose 

Instead of let user or custom guessing why emails are not working it's better to give error to user that the email is not valid. so some solutions I have.
1. Using is_email & not sanitizing. 
2. If you want to sanitize then use `sanitize_email` but also compare the sanitise value with the original if both are diff show user a error it's not valid.

**Comment by singhleo** [comment]
I checked the code we just need to exclude 
```
add_filter( $filter, 'esc_html', 1 );
		add_filter( $filter, 'wp_filter_kses', 2 );
		add_filter( $filter, 'stripslashes', 3 );
```
for filter `bp_get_signup_email_value`. as it's already being sanitized at the bottom. 

![Uploading image.png…]()

@chetansatasiya

**Comment by mikejyoung** [comment]
Hi @tblacker7 

We are investing this further.
Will create a new issue once we have defined the scope of this work and will share the link here.

Thanks,
Mike
Product Manager

CC: @sourabhmatolia
